pkg/virt-operator/install-strategy/strategy.go

```diff
@@ -218,15 +223,26 @@ func dumpInstallStrategyToBytes(strategy *InstallStrategy) []byte {
 	return b.Bytes()
 }
 
+func patchValidation(crd *extv1beta1.CustomResourceDefinition, name string) *extv1beta1.CustomResourceDefinition {
+	if name != "vmi" && name != "vm" {
+		return crd
+	}
+	validation := resources[name]
+	crvalidation := extv1beta1.CustomResourceValidation{}
+	k8syaml.NewYAMLToJSONDecoder(strings.NewReader(validation)).Decode(&crvalidation)
+	crd.Spec.Validation = &crvalidation
+	return crd
+}
+
 func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfig, addMonitorServiceResources bool, operatorNamespace string) (*InstallStrategy, error) {
 
 	strategy := &InstallStrategy{}
 
-	strategy.crds = append(strategy.crds, components.NewVirtualMachineInstanceCrd())
-	strategy.crds = append(strategy.crds, components.NewPresetCrd())
-	strategy.crds = append(strategy.crds, components.NewReplicaSetCrd())
-	strategy.crds = append(strategy.crds, components.NewVirtualMachineCrd())
-	strategy.crds = append(strategy.crds, components.NewVirtualMachineInstanceMigrationCrd())
+	strategy.crds = append(strategy.crds, patchValidation(components.NewVirtualMachineInstanceCrd(), "vmi"))
```

Comment:
Rather than patching the CRDs here in the install-strategy logic, can we add the crd.Spec.Validation in the functions that create the CRDs structs themselves?

for example, for the VMI crd, instead of patching an already created CRD object, the validation could be added to the CRD in the components.NewVirtualMachineInstanceCrd() function.

Reply:
moved and tested